### PR TITLE
Remove Java 8

### LINKED ISSUE #242: Windows tests are failing but build passes
### Version report

Jenkins and plugins versions report:

```
Copy/paste here....
```

- What Operating System are you using (both controller, and any agents involved in the problem)?

```
Paste here
```

### Reproduction steps

- Step 1...
- Step 2...

### Results

Expected result:

Actual result:

### PR DESCRIPTION
**Pull request by slide**
- Remove Java 8 images
- Make Java 11 the default
- Update to newer version of Pester
- Fix make.ps1 to fail the build if the tests fail
- Update windows tests

This first push should fail, will push an update, want to make sure the build fails when it should.
Fixes #242 

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by dduportal — APPROVED** [review]
Thanks!

**Comment by slide** [comment]
It looks like there is a full disk issue on the agents, can we recycle the agents or run a docker system prune on them?

**Comment by slide** [comment]
Close and reopen to get a new build

File: Jenkinsfile [hunk 1]
```diff
@@ -47,6 +47,7 @@ pipeline {
                     post {
                         always {
                             junit(allowEmptyResults: true, keepLongStdio: true, testResults: 'target/**/junit-results.xml')
+                            bat 'docker system prune --force --all || echo "Failed to cleanup docker images"'
```
Comment (timja) [inline Jenkinsfile:50]:
```suggestion
```

File: Jenkinsfile [hunk 2]
```diff
@@ -105,6 +106,7 @@ pipeline {
                     post {
                         always {
                             junit(allowEmptyResults: true, keepLongStdio: true, testResults: 'target/*.xml')
+                            sh 'docker system prune --force --all || echo "Failed to cleanup docker images"'
```
Comment (timja) [inline Jenkinsfile:109]:
```suggestion
```

**Review by timja — COMMENTED** [review]
unsure if we want to add this, better to either recycle in cloud configuration after build or increase disk size so this is less of an issue.

File: Jenkinsfile (hunk 1 above)
Comment (timja) [inline Jenkinsfile:50]:
```suggestion
```

**Comment by slide** [comment]
Still disk full errors:

`[2022-09-22T20:21:14.507Z] failed to register layer: re-exec error: exit status 1: output: BackupWrite \\?\C:\ProgramData\docker\windowsfilter\442605cf2aa7122b550258203052d221da3262fcac05207481ac0d5fa6633b0b\Files\Windows\SysWOW64\twinapi.appcore.dll: There is not enough space on the disk.`

**Comment by timja** [comment]
> Still disk full errors:
> 
> `[2022-09-22T20:21:14.507Z] failed to register layer: re-exec error: exit status 1: output: BackupWrite \\?\C:\ProgramData\docker\windowsfilter\442605cf2aa7122b550258203052d221da3262fcac05207481ac0d5fa6633b0b\Files\Windows\SysWOW64\twinapi.appcore.dll: There is not enough space on the disk.`

I suspect the number of images are too many for the disk? may need multiple agents, like in https://github.com/jenkinsci/docker/blob/master/Jenkinsfile#L72-L87

**Comment by timja** [comment]
That agent currently has 15gb free of a 128gb OS disk

(only 500 IOPS on the disk too)

**Comment by slide** [comment]
Can we prune the existing images on the agent?

**Comment by timja** [comment]
re-triggered now, the windows agents recycle after 30 minutes.

cc @dduportal as an fyi

**Comment by dduportal** [comment]
Last build failed with `"matrix" or "parallel" cannot be nested inside another "matrix" or "parallel"`.

We might need to switch to full scripted syntax (to build a nested matrix) OR to build a top-level matrix with some axis exclusion if we want to stay with declarative.

Also, check https://github.com/jenkinsci/docker-inbound-agent/pull/280

**Comment by slide** [comment]
> We might need to switch to full scripted syntax (to build a nested matrix) OR to build a top-level matrix with some axis exclusion if we want to stay with declarative.
> 

I was hoping to avoid that, but it might be necessary. I saw your PR and will rebase mine on top once yours is merged.

**Review by timja — COMMENTED** [review]
matrix doesn't look quite right:

image

windows and linux for each image

**Comment by slide** [comment]
It's using a when to skip the stage based on the environment variables value

**Comment by timja** [comment]
I think it would be better to separate linux out to it's own stage, e.g. like this:

image

**Comment by slide** [comment]
I was trying to keep it all parallel, but moving Linux out of the matrix is fine.

**Comment by slide** [comment]
This is how it looks now

image

**Review by timja — APPROVED** [review]
> I was trying to keep it all parallel, but moving Linux out of the matrix is fine.

I see, given linux doesn't take that long I think this UX is a lot clearer.

**Pull request merged**